etcdctl/command/util.go

```diff
@@ -211,6 +211,12 @@ func mustNewClient(c *cli.Context) client.Client {
 				handleError(ExitServerError, err)
 			}
 
+			if cerr, ok := err.(*client.ClusterError); ok {
+				if cerr.IsBadConnection() {
+					handleError(ExitBadConnection, err)
+				}
+			}
+
 			// fail-back to try sync cluster with peer API. this is for making etcdctl work with etcd 0.4.x.
 			// TODO: remove this when we deprecate the support for etcd 0.4.
 			eps, serr := syncWithPeerAPI(c, ctx, hc.Endpoints())
```

Comment:
added bad connection check after `syncWithPeerAPI` in case connections went down between first try and fallback